3-enrich/emr-etl-runner/lib/snowplow-emr-etl-runner/emr_config_generator.rb

```diff
@@ -0,0 +1,202 @@
+
+# Author::    Ben Fradet (mailto:support@snowplowanalytics.com)
+# Copyright:: Copyright (c) 2012-2017 Snowplow Analytics Ltd
+# License::   Apache License Version 2.0
+
+require 'contracts'
+require 'iglu-client'
+
+# Implementation of Generator for emr clusters
+module Snowplow
+  module EmrEtlRunner
+    class EmrConfigGenerator
+
+      include Snowplow::EmrEtlRunner::Generator
+      include Snowplow::EmrEtlRunner::Utils
+      include Contracts
+
+      STANDARD_HOSTED_ASSETS = "s3://snowplow-hosted-assets"
+
+      Contract String => Iglu::SchemaKey
+      def get_schema_key(version)
+        Iglu::SchemaKey.parse_key("iglu:com.snowplowanalytics.dataflowrunner/ClusterConfig/avro/#{version}")
+      end
+
+      Contract ConfigHash, Bool, Maybe[String], String, ArrayOf[String] => Hash
+      def create_datum(config, debug=false, resume_from=nil, resolver='', enrichments=[])
+        legacy = (not (config[:aws][:emr][:ami_version] =~ /^[1-3].*/).nil?)
+        region = config[:aws][:emr][:region]
+
+        {
+          "name" => config[:aws][:emr][:jobflow][:job_name],
+          "logUri" => config[:aws][:s3][:buckets][:log],
+          "region" => region,
+          "credentials" => {
+            "accessKeyId" => config[:aws][:access_key_id],
+            "secretAccessKey" => config[:aws][:secret_access_key]
+          },
+          "roles" => {
+            "jobflow" => config[:aws][:emr][:jobflow_role],
+            "service" => config[:aws][:emr][:service_role]
+          },
+          "ec2" => {
+            "amiVersion" => config[:aws][:emr][:ami_version],
+            "keyName" => config[:aws][:emr][:ec2_key_name],
+            "location" => get_location_hash(
+              config[:aws][:emr][:ec2_subnet_id], config[:aws][:emr][:placement]),
+            "instances" => {
+              "master" => { "type" => config[:aws][:emr][:jobflow][:master_instance_type] },
+              "core" => {
+                "type" => config[:aws][:emr][:jobflow][:core_instance_type],
+                "count" => config[:aws][:emr][:jobflow][:core_instance_count],
+                "ebsConfiguration" =>
+                  get_ebs_configuration(config[:aws][:emr][:jobflow][:core_instance_ebs])
+              },
+              "task" => {
+                "type" => config[:aws][:emr][:jobflow][:task_instance_type],
+                "count" => config[:aws][:emr][:jobflow][:task_instance_count],
+                "bid" => config[:aws][:emr][:jobflow][:task_instance_bid].to_s
+              }
+            }
+          },
+          "tags" => get_tags(config[:monitoring][:tags]),
+          "bootstrapActionConfigs" => get_bootstrap_actions(
+            config[:aws][:emr][:bootstrap],
+            config[:collectors][:format],
+            legacy,
+            region,
+            config[:enrich][:versions][:spark_enrich]
+          ),
+          "configurations" => get_configurations(legacy),
+          "applications" => ["Hadoop"]
```

Comment:
There wasn't Spark enrich/shred at the time. Anyway the generate command needs to support r89 and r90.